SDWebImage/Core/UIImage+Metadata.m

```diff
@@ -29,6 +29,27 @@ - (void)setSd_imageLoopCount:(NSUInteger)sd_imageLoopCount {
     objc_setAssociatedObject(self, @selector(sd_imageLoopCount), value, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
 }
 
+- (NSUInteger)sd_imageFrameCount {
+    NSArray<UIImage *> *animatedImages = self.images;
+    if (!animatedImages || animatedImages.count <= 1) {
+        return 1;
+    }
+    __block NSUInteger frameCount = 1;
+    __block UIImage *previousImage = animatedImages.firstObject;
+    [animatedImages enumerateObjectsUsingBlock:^(UIImage * _Nonnull image, NSUInteger idx, BOOL * _Nonnull stop) {
```

Comment:
Since UIImage is const, this calculated loop count can store into image's instance level to avoid re-calculate